### PR TITLE
use string::append to catenate strings

### PR DESCRIPTION
**Pull request by cg31**
With VC++, when std::string is empty, there will be an error -- "string iterator not dereferencable" -- to use string::begin.

And using raw pointer to manipulate std::string doesn't look like a good idea.

### CODE REVIEW

**Comment by concretevitamin** [comment]
Jenkins, test this please.

**Comment by vrv** [comment]
@tensorflow-jenkins test this please

File: google/protobuf [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit caf1fb7197ee94c07108fc7cfbca07432b185a28
```
Comment (vrv) [inline google/protobuf]: you'll want to undo this

**Comment by vrv** [comment]
(the branch has conflicts, can you address the conflict? we can take a look after that)

**Comment by caisq** [comment]
@vrv The conflicts needs to be resolved first before the tests can pass on Jenkins. The base is a few weeks old.

**Comment by vrv** [comment]
My guess is we'll get to this as part of our Windows support.

**Pull request closed without merging**